### ISSUE TITLE
Which firmware for D1 Mini ??

### ISSUE THREAD

**Issue by vincegre** [xoseperez/espurna#474]
Hi,

I'd like to flash a D1 mini board with Espurna but 3 firmwares are avalaible for d1 mini. Which one should i use ?

thanks

Vincèn

**Comment by xoseperez** [comment]
The basic one the "espurna-1.12.1-wemos-d1mini-relayshield.bin", the other two have specific support for power monitor chips you can find in some devices. Check this post if you are interested: http://tinkerman.cat/connected-power-meter/

**Comment by vincegre** [comment]
> The basic one the "espurna-1.12.1-wemos-d1mini-relayshield.bin", the other two have specific support for power monitor chips you can find in some devices.

Thanks for confirmation ;) Flashed with success but not usable in my case as relay is not on the I/O expected and can't change it in hardware (Little piggy modules that pile one above the other....

> Check this post if you are interested: http://tinkerman.cat/connected-power-meter/

Thanks for link but had seen it already ;)

**Comment by xoseperez** [comment]
You can try to build an image with the GPIO for your hardware.

**Comment by chryss44** [comment]
hi, i have choose  espurna-1.12.1-itead-sonoff-t1-2ch.bin for D1 mini pro and it work like a charm :-)

**Comment by vincegre** [comment]
@chryss44 D1 Mini here :/

**Comment by chryss44** [comment]
Yes, i know, and have a Wemos D1 mini pro with espurna-1.12.1-itead-sonoff-t1-2ch on it  and it work

**Comment by chryss44** [comment]
![capture d ecran 2018-01-26 a 13 12 41](https://user-images.githubusercontent.com/32177747/35439357-a6c0a206-029a-11e8-9990-c177bc4d58df.png)
This one

**Comment by vincegre** [comment]
Yep but how did you setup devices connected to it ? as it has to be configured in firmware when you build it :( It can't be configured in live once loaded in D1 !

**Comment by chryss44** [comment]
I prefer setup device D5 for relay 1 and D3 for button 2.

**Comment by chryss44** [comment]
you can use espurna-1.12.1-itead-sonoff-t1-1ch if you need just one relay.

**Comment by vincegre** [comment]
> you can use espurna-1.12.1-itead-sonoff-t1-1ch if you need just one relay.

Not working for me as I expect it :( My relay is only controled if I push second relay in interface and I discovered a bug of the Autodiscover (@xoseperez ) I guess ! It immediately reported that new device in HA as soon as I configured my Espurna D1 but it reports only one relay which is unfortunately for me the first relay in HA (second relay doesn't show unless it should !!

**Comment by chryss44** [comment]
Can you decribe your hardware (relay and button) ?
It is true that I loaded the firmware espurna-1.12.1-itead-sonoff-t1-2ch
and that the first one works, I wanted to use the second later


2018-01-29 10:01 GMT+01:00 Vincèn <notifications@github.com>:

**Comment by vincegre** [comment]
> Can you decribe your hardware (relay and button) ? It is true that I loaded the firmware espurna-1.12.1-itead-sonoff-t1-2ch and that the first one works, I wanted to use the second later

In fact I use these boards https://fr.aliexpress.com/item/15PCS-WAVGAT-D1-mini-Pro-WiFi-development-board-KIT-NodeMcu-Lua-based-on-ESP8266-D1-mini/32816174727.html?spm=a2g0s.9042311.0.0.q60CiM but no worries, I'm going to compile home made version of Espurna for my hardware so I'm sure it works at best for me 👍
